exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/MapBuilder.java

```diff
@@ -123,30 +126,36 @@ public RepeatedListBuilder addRepeatedList(String name) {
     return tupleBuilder.addRepeatedList(this, name);
   }
 
-  private MapColumnMetadata buildCol() {
+  public MapColumnMetadata buildColumn() {
```

Comment:
This was private because only the top-level schema builder was meant to be public. These intermediate items are used internally to create a map in the middle of building a schema.

Might well work by itself, just didn't test it that way. Looks like you've made the parent optional, which might be OK.